src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/SupplementaryRiskService.kt

```diff
@@ -43,6 +44,12 @@ class SupplementaryRiskService(
   fun createNewSupplementaryRisk(supplementaryRiskDto: SupplementaryRiskDto): SupplementaryRiskDto {
     with(supplementaryRiskDto) {
       log.info("Create new supplementary risk for crn: $crn")
+      val existingRisk = sourceId?.let { supplementaryRiskRepository.findBySourceAndSourceId(source.name, it) }
+      if (existingRisk != null)
+        throw DuplicateSourceRecordFound(
+          "Duplicate supplementary found for source: $source with sourceId: $sourceId",
+          existingRisk.toSupplementaryRiskDto("for source: $source and sourceId: $sourceId")
+        )
       return supplementaryRiskRepository.save(this.toSupplementaryRiskEntity())
```

Comment:
why do you prefer this approach checking in each create request vs just checking when a constraint exception is thrown only?